### DISCUSSION TITLE
Total timeout of API google.logging.v2.LoggingServiceV2 exceeded 60000 milliseconds related to Prisma?

### DISCUSSION THREAD

**Discussion by fkubandt** [prisma/prisma#11359] [Q&A]
Hi Prisma Team! 

we want to switch our services (gcp cloud run, PostgreSQL) to use Prisma as ORM. The initial setup was successful: 
pulling and adapting schema, initialising Prisma Client as singleton, adding a test endpoint to connect to the db both locally and within cloudrun via web socket.

However, even though none of our business logic is using Prisma yet, since we set up the testing endpoint (even when not using it) we encounter regular google logging timeouts that lead to container crashes. As soon as we remove any usage of the Prisma Client (but still instantiating it), the problem vanishes. 

Has anyone else ever seen this problem in relation with Prisma? Grateful for any hints, as I would love to continue switching our service to Prisma, but currently we had to roll it back due to the issues. 

We're using a docker container with node:16-buster-slim as base image and libssl1.1. 

Minimal Code example: 
prismaClient.ts
```typescript 
export class MyPrismaClient {
    public static init() {
        return new PrismaClient({
             datasources: { db: { `postgresql://${user}:{password}@127.0.0.1:5432/${dbName}?host=${socketAddress}&schema=public&connection_limit=5`}}
        }
    }
}

public static getInstance(): PrismaClient {
    if (!this.prismaClient) { 
        throw new Error('call init() first');
    }
    return this.prismaClient;
}

```
index.ts
```typescript
const initializeApp = (...) => {
    //...some other initialisations
    MyPrismaClient.init();
    app.listen(port)
} 
const containerStart = () => { initializeApp(routes, routePrefix, port).catch(() => null) }
containerStart()
```

This alone does not lead to any errors. Adding a dummy endpoint
```typescript
    let testRouter = express.Router();
    testRouter.get('/test', async (req, res) => {
    const prisma = MyPrismaClient.Instance;
    const testObject = await Prisma.test.findUnique({
         where: { id: existingId}
    });
   res.status(200).send(testObject);
}
```
even WITHOUT calling it leads to the described errors in gcp.
The docker file is equal in both cases, installing libssl and generating the client.

**Comment by janpio** [comment]
What does `Total timeout of API google.logging.v2.LoggingServiceV2 exceeded 60000 milliseconds` mean in this context? Can you explain a bit?

**Reply by fkubandt** [reply]
Hi Jan, 
yes sure! 

The error states that the google call retry mechanism has failed for 60s while trying to call the google logging api, so some service wants to log but does not get the memory or CPU to do so. Because the logging fails, unfortunately we don't know what the failing log is, the only connection we see is that it only happens only once we use the PrismaClient anywhere in the production code, even when not called (but not when only instantiating it).

I don't see any hints for a memory leak, so my best guess is that Prisma or one of its dependencies causes a log in between or before requests, at a time where we don't have cpu allocated. 


Does Prisma log in the background? Are there any differences in Prisma logging or Prisma using its dependencies at runtime between PrismaClient being instantiated and PrismaClient being used? Do you know of anyone else having had similar issues?

**Reply by janpio** [reply]
Unless you tell it to (via https://pris.ly/d/logging or https://pris.ly/d/debugging) Prisma Client and its dependencies are completely silent if there are no errors.

**Reply by fkubandt** [reply]
Hi Jan, 
thanks for the reply but I am not sure I understand it. Could you please rephrase it?

**Reply by janpio** [reply]
Ugh, that looked like I had written it with auto completion. Fixed the wording now.

**Reply by vladi-stevanovic** [reply]
👋 Hello @fkubandt I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
